content/cybersecurity/concepts/nmap/terms/nmap-scripting-engine/nmap-scripting-engine.md

```diff
@@ -0,0 +1,52 @@
+---
+Title: 'Nmap Scripting Engine'
+Description: 'Allows for the writing of scripts that automate a wide variety of tasks.'
+Subjects:
+  - 'Computer Science'
+  - 'Networking'
+Tags:
+  - 'Networking'
+  - 'Cybersecurity'
+  - 'Ethical Hacking'
+CatalogContent:
+  - 'introduction-to-cybersecurity'
+  - 'paths/scan-systems-with-nmap'
+---
+
+The **Nmap Scripting Engine (NSE)** allows the writing of scripts that automate a wide variety of tasks. As one of the more powerful and flexible features of Nmap, it enables the automation of networking tasks and allows the script to be shared between users.
+
+The embedded Lua programming language is used to write scripts for Nmap. The reference manual for Lua version 5.3 is available [freely online](http://www.lua.org/manual/5.3/).
+
+The NSE is activated with the `-sC` option. The `--script` option can be used to specify a custom set of scripts. The results are integrated into Nmap's output.
+
+> **Note:** Scripts should never be run from untrusted third parties, as they can accidentally or maliciously damage the system or invade the user's privacy.
+
+## Options
+
+The following options are related to NSE:
+
+- `-sC`: Enables the most common scripts.
+- `--script`: Chooses scripts by category, directory, or individual script filename.
+- `--script-args` and `script-args-file`: Provides arguments to scripts that use/require them.
+- `--script-help`: Shows a description of what each selected script does.
+
+Scripts are generally run with a port scan because the state of each port may determine if a script runs or not. It is possible to run scripts without a port scan, using only host discovery, with the `-sn` option. In this case, only host scripts will run. To run without a port scan or host discovery, use `-Pn -sn`. Every host will be assumed up and only host scripts will run.
```

Comment:
```suggestion
Scripts are generally run with a port scan because the state of each port may determine if a script runs or not. It is possible to run scripts without a port scan, using only host discovery, with the `-sn` option. In this case, only host scripts will run. The `-Pn -sn` flag is used to run without a port scan or host discovery. Every host will be assumed up and only host scripts will run.
```